otelcol/config.go

```diff
@@ -60,6 +61,9 @@ type Config struct {
 	Extensions map[component.ID]component.Config
 
 	Service service.Config
+
+	// Conf is the confmap.Conf used to create this Config
+	Conf *confmap.Conf
```

Comment:
Not ok, this is not parsed from the config, you break the design here since everything else is parsed from the config.

Reply:
I agree, it's not ideal. I've pushed a commit that makes this field a method to separate it from the public fields on this struct. Let me know what you think.

If we don't want the source `confmap.Conf` struct available from `Config`, I see two possibilities:
* We could change the `ConfigProvider.Get` method to also return `*confmap.Conf`. To me, this muddies the method signature a little bit, but would provide a clear separation between the two types.
* We could add a new method to `ConfigProvider` to only resolve and return a `confmap.Conf` struct. I opted against this since we are already resolving the config in `Get` and I think another method would be unnecessary on the `ConfigProvider` interface, but this would also allow clearly separating `confmap.Conf` and `Config`.

Let me know if there's another option I've missed.

Reply:
@bogdandrutu I'm sorry, I think I misunderstood what you meant.

If we are looking to have both `Config` and `ConfigProvider` be agnostic of the underlying configuration structure, then I see a few options:

* Provide an optional interface for config providers to implement that resolves the configuration and translates it (if necessary) to a `confmap.Conf` object. This seems like it would probably be the most straightforward solution.
* Provide another type that can be marshaled into YAML from either the `Config` or `ConfigProvider` types that is used in `extension.ConfigWatcher`, in place of `confmap.Conf`. 
* Provide a way to translate `Config` instances back into `confmap.Conf`.

What do you think?

Reply:
@bogdandrutu I've pushed a commit including a rough draft for an optional interface to provide `confmap.Conf` objects from a Config Provider. Let me know what you think.